### ISSUE TITLE
Fill required attributes for a block

### ISSUE THREAD

**Issue by radeksimko** [hashicorp/vscode-terraform#719]
## Problem Statement

Users can today introduce a new block into their configuration via completion

![2021-08-05 13 02 09](https://user-images.githubusercontent.com/287584/128346629-f3f95314-adfa-4fc4-b473-acf303ca4e04.gif)

However the block is not considered complete/valid until it has all required attributes and blocks.
This means that users have to either scrolls over a potentially long list of attributes and blocks to add all required ones and/or consult documentation in a separate window.

It would be great if all required attributes and blocks were pre-filled for them.

### New Block

Upon confirmation of label completion - such as resource type

![Screenshot 2021-08-05 at 14 50 36](https://user-images.githubusercontent.com/287584/128361473-cbe64a4a-c43d-4a2b-b439-db577fb15d47.png)

the editor would also pre-fill all required attributes and blocks

![Screenshot 2021-08-05 at 14 54 06](https://user-images.githubusercontent.com/287584/128362027-fbdff0ca-a199-4a83-abb3-e5f4eb614d3c.png)

### Block with existing attributes/blocks

A contextual code action would be available to the user to fill any _missing_ required attributes and blocks, similar to gopls:

![126799179-dd5a7450-8033-462a-aee3-92b71fa638c9](https://user-images.githubusercontent.com/287584/128362306-d7e1f051-83f7-4e29-ae8b-d7220d50a767.gif)

### Proposal

 - https://github.com/hashicorp/terraform-ls/issues/96

**Comment by jpogran** [comment]
We were able to modify `labelCandidatesFromDependentSchema` to find all required attributes for the selected type and construct an `AdditionalTextEdit` array, however `AddtionalTextEdit` [does not appear](https://github.com/microsoft/language-server-protocol/issues/234) to support snippet completion syntax. This means that even though we can build the list of additional snippets, it won't be used for the main snippet completion

We abandoned that approach and instead moved to build a complete snippet with both the type selected and all required attributes and use that as the main TextEdit, in effect replacing the existing text the user just typed. This succeeded in the sense that the correct information was produced but failed in the sense that it was not displayed client side. Further investigation surmised that the additional content for the attributes exceeded the amount of data that could be received and displayed by the client.

This led us to the LSP [resolve completion method](https://microsoft.github.io/language-server-protocol/specifications/specification-current/#completionItem_resolve). This would allow the user to received a list of types to select, then select the type, then receive only the required attributes for that type. This requires implementing a new request handler and parsing infrastructure, with some new design considerations, which is out of scope for this ticket. Once that work is complete, this ticket will be picked back up.

**Comment by 770715** [comment]
That would be a killer feature. It is implemented in the TF plugin for IntelliJ, and it's the only reason why I use IJ.
